salt/fileserver/__init__.py

```diff
@@ -757,6 +757,36 @@ def file_list(self, load):
             ret = [f for f in ret if f.startswith(prefix)]
         return sorted(ret)
 
+
+    def file_stats(self, load):
+        '''
+        Return a list of files from the dominant environment
+        '''
+        ret = {}
+        if 'saltenv' not in load:
+            return []
+        if not isinstance(load['saltenv'], six.string_types):
+            load['saltenv'] = six.text_type(load['saltenv'])
+
+        for fsb in self._gen_back(load.pop('fsbackend', None)):
+            fstr = '{0}.file_stats'.format(fsb)
+            if fstr in self.servers:
+                ret.update(self.servers[fstr](load)) # merging dicts here
+            else:
+                # if any of used fs backend doesn't support new file_stats
+                # we return False, so that client can use old _file_find function
+                # gitfs is not yet supported
+                return False
```

Comment:
Yeah actually that approach isn't 100% correct, because my list comprehension doesn't check for membership in ``self.servers``.